src/ray/core_worker/reference_count.h

```diff
@@ -106,42 +187,70 @@ class ReferenceCounter {
   std::unordered_map<ObjectID, std::pair<size_t, size_t>> GetAllReferenceCounts() const
       LOCKS_EXCLUDED(mutex_);
 
+  ReferenceTable PopBorrowerRefs(const ObjectID &object_id);
+  void PopBorrowerRefsPointer(const ObjectID &object_id,
+                              ReferenceCounter::ReferenceTable *refs);
+
+  void WrapObjectId(const ObjectID &object_id, const std::vector<ObjectID> &inner_ids,
+                    const absl::optional<rpc::WorkerAddress> &owner_address)
```

Comment:
Instead of adding a new method, why not combine it with existing methods to avoid misuse?

Reply:
It's not possible to do for the "task returns ID" case, since there is no ReferenceCounter method that is called for that right now.

Reply:
Ah I see, yes, we can do it for that case (ray.put). But we will need to keep it as a public method so that we can call it for the case that I mentioned above.

It is also slightly ugly because both WrapObjectId and AddOwnedObject will have to acquire a lock, but I can add an internal version of WrapObjectId.